new-frontiers-14/frontier-station-14
# Cryosleep Fallbacks

## About the PR



Adds a component for fallback cryosleep pods, an entity that automatically comes marked as a fallback cryopod, and replaces all cryopods mapped on Frontier Outpost with these fallback versions.

In the Cryosleep system, if a request is made to wake up, but the user's cryopod has been removed or destroyed, we will check for any available (unoccupied) fallback cryo pod, i.e. those at Frontier Outpost, where they will wake up.

In the case that the original cryopod a user slept at is still available, but currently occupied, they will have to wait.  This helps avoid issues with free teleports with equipped gear.

## Why / Balance


#2476.  A character being round removed unnecessarily sucks.

## How to test


1. Open two clients.
2. On Client 1, purchase a ship.
3. On Client 2, walk onto the ship, place a cryo pod, sleep in it.
4. On Client 1, ensure there is a fallback cryo pod (if in the dev map, you'll need to place one).
5. On Client 1, sell the ship.
6. On Client 2, try to uncryo, you should arrive at the cryo pod.

You can also test the case where your pod is occupied:
7. On Client 2, sleep in the fallback cryo pod.
8. On Client 1, enter the cryo pod.
9. On Client 2, try to uncryo, you will be notified that the pod is occupied.

## Media


## Requirements



## Breaking changes


**Changelog**


:cl:
- fix: When returning from cryosleep, if your pod has been destroyed, you will wake up at Frontier Outpost.
